pkg/sloop/queries/respayloadquery.go

```diff
@@ -40,32 +46,40 @@ func GetResPayload(params url.Values, t typed.Tables, startTime time.Time, endTi
 		var stats typed.RangeReadStats
 
 		keyComparator := getKeyComparator(params)
+		glog.V(glogVerbose).Infof("GetResPayload: keyComparator: %v", keyComparator.String())
 		valPredFn := typed.KubeWatchResult_ValPredicateFns(isResPayloadInTimeRange(startTime, endTime))
 
 		var rangeReadErr error
 		watchRes, _, rangeReadErr = t.WatchTable().RangeRead(txn, keyComparator, nil, valPredFn, startTime, endTime)
 		if rangeReadErr != nil {
+			glog.V(glogVerbose).Infof("GetResPayload: range read error: %v", rangeReadErr)
 			return rangeReadErr
 		}
+		glog.V(glogVerbose).Infof("GetResPayload: range read found: %v payload", len(watchRes))
 
 		// get the previous key for those who has same payload but just before startTime
 		var getPreviousErr error
 		seekKey := getSeekKey(keyComparator, startTime)
+		glog.V(glogVerbose).Infof("GetResPayload: seekKey: %v", seekKey.String())
 		previousKey, getPreviousErr = t.WatchTable().GetPreviousKey(txn, seekKey, keyComparator)
 
 		// when getPreviousErr is not nil, we will not return err since it is ok we did not find previous key from startTime,
 		// we can continue using the result from rangeRead to proceed the rest payload
 		if getPreviousErr == nil {
+			glog.V(glogVerbose).Infof("GetResPayload: previousKey: %v", previousKey.String())
```

Comment:
isn't previousKey the same as seekKey?

Reply:
it might be. seekKey is where to start looking. previousKey is the first key found.
Since seekKey is set with the timeStamp & partitionId based on the range read start time (based on where the user clicked), it seems unlikely